ci/pipelines/deploy.yml

```diff
@@ -0,0 +1,162 @@
+      aws_access_key_id: ((deploy-access-key-id))
+      aws_secret_access_key: ((deploy-secret-access-key))
+
+  - name: runner
+    # See https://github.com/alphagov/govwifi-concourse-runner for a reference dockerfile
+    # readonly_private_ecr_repo_url is provided by the hosted Concourse
+    type: docker-image
+    source:
+      repository: "((readonly_private_ecr_repo_url))"
+      tag: concourse-runner-latest
+
+  - name: mysql-image
+    type: registry-image
+    source:
+      repository: mysql
+      tag: '5.7'
+
+  - name: ruby-image
+    type: registry-image
+    source:
+      repository: ruby
+      tag: '2.6.2'
+
+  - name: nginx-image
+    type: registry-image
+    source:
+      repository: nginx
+      tag: alpine
+
+groups:
+  - name: test
+    jobs:
+      - lint & test
+  - name: push
+    jobs:
+      - Push Image
+  - name: staging
+    jobs:
+      - migrate staging
+      - deploy staging
+  - name: production
+    jobs:
+      - production gate
+      - migrate production
+      - deploy production
+
+jobs:
+  - name: lint & test
+    plan:
+      - aggregate:
+        - get: govwifi-admin
+          trigger: true
+        - get: mysql-image
+          params: {format: oci}
+        - get: ruby-image
+          params: {format: oci}
+        - get: nginx-image
+          params: {format: oci}
+        - get: runner
+      - task: pre-build
+        privileged: true
+        file: govwifi-admin/ci/tasks/pre-build.yml
+        image: runner
+      - aggregate:
+        - task: lint
+          privileged: true
+          file: govwifi-admin/ci/tasks/lint.yml
+          image: runner
+        - task: test
+          privileged: true
+          file: govwifi-admin/ci/tasks/test.yml
+          image: runner
+
+  - name: Push Image
+    plan:
+      - get: govwifi-admin
+        trigger: true
+        passed: ['lint & test']
+      - put: ecr
+        params:
+          build: govwifi-admin
+          build_args:
+            BUNDLE_INSTALL_FLAGS: '--without test development'
+
+  - name: migrate staging
+    plan:
+      - get: govwifi-admin
+        trigger: true
+        passed: ['Push Image']
+      - get: runner
+      - task: 'migrate'
+        file: govwifi-admin/ci/tasks/migrate.yml
+        image: runner
+        params:
+          DEPLOY_STAGE: staging
+          AWS_ACCESS_KEY_ID: ((deploy-access-key-id))
+          AWS_SECRET_ACCESS_KEY: ((deploy-secret-access-key))
+          AWS_DEFAULT_REGION: ((deploy-region))
+
+  - name: deploy staging
+    plan:
+      - get: govwifi-admin
+        passed: [migrate staging]
+        trigger: true
+      - get: runner
+      - task: deploy staging
+        image: runner
+        file: govwifi-admin/ci/tasks/deploy.yml
+        params:
+          AWS_ACCESS_KEY_ID: ((deploy-access-key-id))
+          AWS_DEFAULT_REGION: ((deploy-region))
+          AWS_SECRET_ACCESS_KEY: ((deploy-secret-access-key))
+          DEPLOY_STAGE: staging
+
+  - name: production gate
```

Comment:
What does gate refer to here?

Reply:
This is the manual step for triggering production deploys.
Can rename to `Confirm deploy to production`, which is the same as Jenkins?

Reply:
Yeah maybe? Gate doesn't mean much to me. Is it some Concourse specific concept?